standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2223,6 +2224,29 @@ public PartitionsResponse getPartitionsRequest(PartitionsRequest req)
     return deepCopyPartitions(FilterUtils.filterPartitionsIfEnabled(isClientFilterEnabled, filterHook, parts));
   }
 
+  @Override
+  public GetPartitionsByNamesResult getPartitionsByNames(GetPartitionsByNamesRequest req)
+          throws NoSuchObjectException, MetaException, TException {
+    checkDbAndTableFilters(req.getCatName(), req.getDb_name(), req.getTbl_name());
+    req.setDb_name(prependCatalogToDbName(req.getCatName(), req.getDb_name(), conf));
+    if (req.getValidWriteIdList() == null) {
+      req.setValidWriteIdList(getValidWriteIdList(prependCatalogToDbName(req.getCatName(), req.getDb_name(),
+              conf), req.getTbl_name()));
+    }
+    if (req.getId() <= 0) {
+      req.setId(getTable(prependCatalogToDbName(req.getCatName(), req.getDb_name(), conf), req.getTbl_name()).getId());
```

Comment:
The table ID here will be only used by the cached HMS client, right? If it is used by HiveMetastoreClient, will we throw an error if the ID from getTable different from the one is passed-in?

Reply:
That's right.

Reply:
getTable put here will add one more calls to HMS if cache is not used. So it should be at cachedHMS client.

Reply:
Please check now. I have removed this logic from HiveMetaStoreClient.java .